velox/connectors/hive/HiveConnectorUtil.cpp

```diff
@@ -569,13 +569,15 @@ bool applyPartitionFilter(
 } // namespace
 
 bool testFilters(
-    common::ScanSpec* scanSpec,
-    dwio::common::Reader* reader,
+    const common::ScanSpec* scanSpec,
+    const dwio::common::Reader* reader,
     const std::string& filePath,
     const std::unordered_map<std::string, std::optional<std::string>>&
         partitionKey,
-    std::unordered_map<std::string, std::shared_ptr<HiveColumnHandle>>*
+    const std::unordered_map<std::string, std::shared_ptr<HiveColumnHandle>>&
         partitionKeysHandle) {
+  VELOX_CHECK_EQ(partitionKey.size(), partitionKeysHandle.size());
```

Comment:
After this PR was merged, we encountered some issues accessing the Iceberg table in Gluten. We have some case `partitionKey` of size 1, while `partitionKeysHandle` is empty. When querying the Iceberg table and the columns do not include the partition columns, the `partitionKeysHandle` does not have the partition columns, while the `partitionKey` comes from the table schema and always includes the partition columns.
```
- TPC-H q2 *** FAILED ***
2024-04-11T04:56:12.3100998Z   org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 38.0 failed 1 times, most recent failure: Lost task 0.0 in stage 38.0 (TID 1881) (6497b7236305 executor driver): org.apache.gluten.exception.GlutenException: java.lang.RuntimeException: Exception: VeloxRuntimeError
2024-04-11T04:56:12.3103588Z Error Source: RUNTIME
2024-04-11T04:56:12.3104054Z Error Code: INVALID_STATE
2024-04-11T04:56:12.3104494Z Reason: (1 vs. 0)
2024-04-11T04:56:12.3104830Z Retriable: False
2024-04-11T04:56:12.3105386Z Expression: partitionKey.size() == partitionKeysHandle.size()
2024-04-11T04:56:12.3108115Z Context: Split [Hive: file:/__w/incubator-gluten/incubator-gluten/gluten-iceberg/target/scala-2.12/test-classes/tpch-data-iceberg-velox/default/part/data/p_brand=Brand%2313/00000-2-eba45061-8a65-4cf0-b705-e731e6e86a9a-00004.parquet 4 - 3136] Task Gluten_Stage_38_TID_1881
2024-04-11T04:56:12.3110247Z Top-Level Context: Same as context.
2024-04-11T04:56:12.3110714Z Function: testFilters
2024-04-11T04:56:12.3111921Z File: /__w/incubator-gluten/incubator-gluten/ep/build-velox/build/velox_ep/velox/connectors/hive/HiveConnectorUtil.cpp
2024-04-11T04:56:12.3112976Z Line: 579
```